addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3434,20 +3436,26 @@ registry.ImageOptimize = ImageHandlerOption.extend({
      * @override
      */
     _computeMaxDisplayWidth() {
-        // TODO: read widths from computed style in case container widths are not default
-        const displayWidth = this._getImg().clientWidth;
-        // If the image is in a column, it might get bigger on smaller screens.
-        // We use col-lg for this in snippets, so they get bigger on the md breakpoint
-        if (this.$target.closest('[class*="col-lg"]').length) {
-            // container and o_container_small have maximum inner width of 690px on the md breakpoint
+        const computedStyles = window.getComputedStyle(this._getImg());
+        const displayWidth = parseInt(computedStyles.width);
+
+        // For the logos we don't want to suggest a width too small
+        if (this.$target.closest('nav').length) {
+            return displayWidth * 3;
+        } else if (this.$target.closest('[class*="col-lg"]').length) {
             if (this.$target.closest('.container, .o_container_small').length) {
-                return Math.min(1920, Math.max(displayWidth, 690));
+                const containerMdMaxWidth = parseInt(computedStyles.getPropertyValue('--o-container-md-max-width'));
+                return Math.min(this.MAX_SUGGESTED_WIDTH, Math.max(displayWidth, containerMdMaxWidth));
             }
-            // A container-fluid's max inner width is 962px on the md breakpoint
-            return Math.min(1920, Math.max(displayWidth, 962));
+
+            const containerLgMaxWidth = parseInt(computedStyles.getPropertyValue('--o-container-lg-max-width'));
+            return Math.min(this.MAX_SUGGESTED_WIDTH, Math.max(displayWidth, containerLgMaxWidth));
+        } else if (this.$target.closest('.container, .o_container_small, .container-fluid').length) {
+            const containerXlMaxWidth = parseInt(computedStyles.getPropertyValue('--o-container-xl-max-width'));
+            return Math.min(this.MAX_SUGGESTED_WIDTH, Math.max(displayWidth, containerXlMaxWidth));
```

Comment:
I am kinda confused by this whole method... it needs a comment at each return to explain why we choose this value exactly. I am not even sure how this value was chosen anymore :grimacing: